modules/lss-creditserweiterung/creditserweiterung.user.js

```diff
@@ -0,0 +1,220 @@
+}
+var earnedCredits = "undefined";
+var creditsToNextRank = "undefined";
+
+function setCookies() {
+  earnedCredits = NaN;
+  rank = NaN;
+  content = $(".lightbox_iframe");
+  content = content.first()[0];
+  content = content.contentWindow.document;
+  content = content.children[0].children[1].children[2].children[0];
+  content = content.innerHTML;
+  content = content.replace(I18n.t('lssm.creditserweiterung.setCookieRegex1'), "");
+  content = content.replace(I18n.t('lssm.creditserweiterung.setCookieRegex2'), "");
+  content = content.replace(I18n.t('lssm.creditserweiterung.setCookieRegex3'), "");
+  earnedCredits = content.replace(I18n.t('lssm.creditserweiterung.setCookieRegex4'), "");
+  earnedCredits = earnedCredits.replace(I18n.t('lssm.creditserweiterung.setCookieRegex5'), "");
+  rank = content.replace(I18n.t('lssm.creditserweiterung.setCookieRegex6'), "");
+  document.cookie = I18n.t('lssm.creditserweiterung.earnedCreditsCookie') + "=" + earnedCredits + "; path=/";
+  document.cookie = I18n.t('lssm.creditserweiterung.rankCookie') + "=" + rank + "; path=/";
+  console.log("Saved Cookies!");
+}
+
+function readCookie(name) {
+    var nameEQ = name + "=";
+    var ca = document.cookie.split(';');
+    for(var i=0;i < ca.length;i++) {
+        var c = ca[i];
+        while (c.charAt(0)==' ') c = c.substring(1,c.length);
+        if (c.indexOf(nameEQ) === 0) return c.substring(nameEQ.length,c.length);
+    }
+    return null;
+}
+
+function fnct_nextRank(){
+    var earnedCredits = parseInt(readCookie(I18n.t('lssm.creditserweiterung.earnedCreditsCookie')));
+    for (i=0; I18n.t('lssm.creditserweiterung.creditsNeeded')[i]<earnedCredits; i++){
+        var nextRank = (I18n.t('lssm.creditserweiterung.ranks')[I18n.t('lssm.creditserweiterung.creditsNeeded')[i+1]]);
+    }
+    return nextRank;
+}
+
+function fnct_nextCreditsNeeded(){
+  var earnedCredits = parseInt(readCookie(I18n.t('lssm.creditserweiterung.earnedCreditsCookie')));
+  for (i=0; I18n.t('lssm.creditserweiterung.creditsNeeded')[i]<earnedCredits; i++){
+      var neededCredits = I18n.t('lssm.creditserweiterung.creditsNeeded')[i+1];
+  }
+  return neededCredits;
+}
+
+function lightbox_close(){
+    $("#lightbox_close").click();
+}
+
+lightboxOpen(src="/profile/"+user_id);
+window.setTimeout(setCookies, 1000);
+window.setTimeout(lightbox_close, 3000);
+
+creditsEarned = parseInt(readCookie(I18n.t('lssm.creditserweiterung.earnedCreditsCookie'))).toLocaleString();
+nextRank = fnct_nextRank();
+needed = fnct_nextCreditsNeeded();
+earnedCredits = readCookie(I18n.t('lssm.creditserweiterung.earnedCreditsCookie'));
+rest = needed - earnedCredits;
+creditsToNextRank = parseInt(rest).toLocaleString();
+if (creditsEarned == "NaN"){
+    creditsEarned = I18n.t('lssm.creditserweiterung.reloadMessage');
+}
+if (creditsToNextRank == "NaN"){
+    creditsToNextRank = I18n.t('lssm.creditserweiterung.reloadMessage');
+}
+if (I18n.t('lssm.creditserweiterung.coinsProtokoll') != "undefined"){
```

Comment:
Als Entwickler, der diesen Code zum ersten Mal sieht, war es mir nicht möglich auf Anhieb zu erkennen was genau diese beiden IF Blöcke machen, bzw. wann der eine und wann der andere ausgeführt wird. Genau das sollte jedoch dein Anspruch sein, um die Wartbarkeit des Codes zu gewährleisten.

Die Vorgehensweise mit dem "undefined" mag technisch offenbar funktionieren, ist aber ein Anti-Pattern wie es im Bilderbuch steht.

Außerdem habe ich hier noch zu bemängeln, dass es in den beiden IF Blöcken sehr viel redundanten Code gibt. Das gilt es immer zu vermeiden.
Ziehe das allgemeine nach außen, damit du es in den einzelnen Bedingungen nicht mehrfach deklarieren musst.

Reply:
Das hatte ich ursprünglich gemacht, aber dann wars nichtmehr in der Dropdown, ich hoffe, ass ich da einfach nur einen kleinen Fehler drin hatte, kümmer ich mich drum